src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Directives.cs

```diff
@@ -318,8 +331,10 @@ private void TagHelperDirective(string keyword, Func<string, ISpanCodeGenerator>
                 if (!startsWithQuote ||
                     !rawValue.EndsWith("\"", StringComparison.OrdinalIgnoreCase))
                 {
-                    Context.OnError(startLocation,
-                                    RazorResources.FormatParseError_DirectiveMustBeSurroundedByQuotes(keyword));
+                    Context.OnError(
+                        startLocation,
+                        RazorResources.FormatParseError_DirectiveMustBeSurroundedByQuotes(keyword),
+                        keywordWithWhitespace.Length + rawValue.Length);
```

Comment:
The length here will extend past the end of the directive value due to the `startLocation` occurring here: `@addTagHelper |`.

For this error I wouldn't use the length of the `keyword + rawValue.Length` anyways though. It's referencing the value of the directive being an invalid format. So for: `@addTagHelper "*` I'd highlight the `"*`